File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/abstractClass/abstractExtendsNonAbstract/AnimalMain.java
package de.fraunhofer.iem.classesAndObjects.classes.abstractClass.abstractExtendsNonAbstract;

/**
 * Demonstrates the usage of the abstract class that extends the non-abstract class
 *
 * @author Ranjith Krishnamurthy
 */
public class AnimalMain {
    public static void main(String[] args) {
        Animal animal = new Cat("AAATGC");
        animal.makeNoise();
        animal.displayDNA();
        new NoiseBox().makeNoise();
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/abstractClass/abstractExtendsNonAbstract/AnimalMain.kt
package de.fraunhofer.iem.classesAndObjects.classes.abstractClass.abstractExtendsNonAbstract

/**
 * Demonstrates the usage of the abstract class that extends the non-abstract class
 *
 * @author Ranjith Krishnamurthy
 */
fun main() {
    val animal: Animal = Cat("AAATGC")
    animal.makeNoise()
    animal.displayDNA()
    NoiseBox().makeNoise()
}